lib_eio_posix/fd.ml

```diff
@@ -0,0 +1,60 @@
+open Eio.Std
+
+module Rcfd = Eio_unix.Private.Rcfd
+
+type t = {
+  fd : Rcfd.t;
+
+  (* stdin, stdout and stderr are blocking, and so need special care.
+     For these, we first wait for them to become e.g. readable and then hope
+     that the read doesn't block. This may fail if multiple fibers try to read
+     at the same time. We could check that it's still readable after being
+     resumed, but that still won't work if multiple domains read at the same
+     time. Same problem for writes. *)
+  blocking : bool;
```

Comment:
I think it might be worth considering making stdin nonblocking, and being careful to restore the blockiness on fork. 
This way at least we guarantee it's safe (even if silly) to have two fibers waiting on input.
I'd not touch stderr or stdout, because users will still end up using Printf directly, but stdin we might consider assuming "this is ours". At any rate would be nice to see what others are doing, maybe the nonblock->restore dance ?

Reply:
Lwt does the same thing (waits and then does the read): https://github.com/ocsigen/lwt/blob/cc05e2bda6c34126a3fd8d150ee7cddb3b8a440b/src/unix/lwt_unix.cppo.ml#L656

But it does it in a systhread, so it's not a big problem. We can do the same once eio_posix uses systhreads (after this initial PR is merged).